cylc/flow/config_schema.py

```diff
@@ -0,0 +1,642 @@
+                'warning handler': [VDR.V_STRING_LIST, None],
+                'critical handler': [VDR.V_STRING_LIST, None],
+                'retry handler': [VDR.V_STRING_LIST, None],
+                'submission retry handler': [VDR.V_STRING_LIST, None],
+                'execution timeout handler': [VDR.V_STRING_LIST, None],
+                'submission timeout handler': [VDR.V_STRING_LIST, None],
+                'custom handler': [VDR.V_STRING_LIST, None],
+            },
+            'suite state polling': {
+                'user': [VDR.V_STRING],
+                'host': [VDR.V_STRING],
+                'interval': [VDR.V_INTERVAL],
+                'max-polls': [VDR.V_INTEGER],
+                'message': [VDR.V_STRING],
+                'run-dir': [VDR.V_STRING],
+                'verbose mode': [VDR.V_BOOLEAN],
+            },
+            'environment': {
+                '__MANY__': [VDR.V_STRING],
+            },
+            'directives': {
+                '__MANY__': [VDR.V_STRING],
+            },
+            'outputs': {
+                '__MANY__': [VDR.V_STRING],
+            },
+            'parameter environment templates': {
+                '__MANY__': [VDR.V_STRING],
+            },
+        },
+    },
+    'suite run platforms': {
+        'run hosts': [VDR.V_SPACELESS_STRING_LIST],
+        'run ports': [VDR.V_INTEGER_LIST, list(range(43001, 43101))],
+        'condemned hosts': [VDR.V_ABSOLUTE_HOST_LIST],
+        'auto restart delay': [VDR.V_INTERVAL],
+        'run host select': {
+            'rank': [VDR.V_STRING, 'random', 'load:1', 'load:5', 'load:15',
+                     'memory', 'disk-space'],
+            'thresholds': [VDR.V_STRING],
+        },
+        'suite host self-identification': {
+            'method': [VDR.V_STRING, 'name', 'address', 'hardwired'],
+            'target': [VDR.V_STRING, 'google.com'],
+            'host': [VDR.V_STRING],
+        },
+    },
+    'visualization': {
+        'initial cycle point': [VDR.V_CYCLE_POINT],
+        'final cycle point': [VDR.V_STRING],
+        'number of cycle points': [VDR.V_INTEGER, 3],
+        'collapsed families': [VDR.V_STRING_LIST],
+        'use node color for edges': [VDR.V_BOOLEAN],
+        'use node fillcolor for edges': [VDR.V_BOOLEAN],
+        'use node color for labels': [VDR.V_BOOLEAN],
+        'node penwidth': [VDR.V_INTEGER, 2],
+        'edge penwidth': [VDR.V_INTEGER, 2],
+        'default node attributes': [
+            VDR.V_STRING_LIST, ['style=unfilled', 'shape=ellipse']],
+        'default edge attributes': [VDR.V_STRING_LIST],
+        'node groups': {
+            '__MANY__': [VDR.V_STRING_LIST],
+        },
+        'node attributes': {
+            '__MANY__': [VDR.V_STRING_LIST],
+        },
+    },
+}
+# where:
+# - value_type: value type (compulsory).
+# - default: the default value (optional).
+# - allowed_2, ...: the only other allowed values of this setting (optional).
+
+
+def upg(cfg, descr):
```

Comment:
Note to self - create a ticket for updating the validation tests and the more complicated config schema.